### PR TITLE
Changed pre-build job definition and renderer

### PR DESCRIPTION
**Pull request by bamachrn**
* pre-build job generationupdated as jenkins is not able to read file from out side of workspace
* updated jenkins job config for enabling git push to rendered repository

### CODE REVIEW

File: pre_build/cccp_index_pre_build_check.py [hunk 1]
```diff
@@ -116,48 +116,44 @@ def main(indexdlocation):
...
-                t = tempfile.mkdtemp()
-                generated_filename = os.path.join(
-                    t,
-                    'cccp_GENERATED.yaml'
-                )
+                pre_text = ''.join(random.sample(
+                    string.lowercase + string.digits, 10))
+                generated_filename = '_'.join(pre_text,
+                                              'cccp_GENERATED.yaml'
+                                              )
```
Comment (navidshaikh) [inline pre_build/cccp_index_pre_build_check.py]: so `randomString_cccp_GENERATED.yml` is going to be inside workspace directory and not in temporary directory?

File: pre_build/pre-build-job.yml [hunk 2]
```diff
@@ -57,18 +54,22 @@
...
-              "git add -A && "
-              "git commit -am 'Added changes from pre-build' && "
-              "git push prebuild-render {prebuild_branch} && "
+              $ssh_cmd -t "cd payload && rm -rf .gitignore .git .gitkeep||true && \
+              git init && git config --global user.name 'Container Pipeline' && \
+              git config --global user.email container@centos.org && \
+              git remote add prebuild-render {prebuild_render} && \
+              git checkout -b {prebuild_branch} && \
+              git add -A && \
+              git commit -am $GIT_COMMIT && \
```
Comment (navidshaikh) [inline pre_build/pre-build-job.yml]: what will be the value of `$GIT_COMMIT` or where it is declared/defined?

File: pre_build/cccp_index_pre_build_check.py (hunk 1 above)
Comment (bamachrn) [reply pre_build/cccp_index_pre_build_check.py]: yup, and every run cleans the workspace.

File: pre_build/pre-build-job.yml (hunk 2 above)
Comment (bamachrn) [reply pre_build/pre-build-job.yml]: this is jenkins variable commit id from the source repo. we can make use of it for build reason in next iteration

**Pull request merged**